roles/calico/README.rst

```diff
@@ -0,0 +1,147 @@
+Calico
+======
+
+.. versionadded:: 0.4
+
+`Calico <http://www.projectcalico.org>`_ is used in the project to add the IP
+per container functionality. Calico connects Docker containers through IP no matter
+which worker node they are on. Calico uses :doc:`etcd` to distribute information
+about workloads, endpoints, and policy to each worker node. Endpoints are
+network interfaces associated with workloads. Calico is deployed in the Docker
+container on each worker node and managed by systemd. Any workload managed by
+Calico is registered as a service in Consul.
+
+Modes
+^^^^^
+
+Calico can run a public cloud environment that does not allow either L3 peering
+or L2 connectivity between Calico hosts. Calico will then route traffic between
+the Calico hosts using IP in IP mode. At this time, the full node-to-node BGP
+mesh is supported and configured in CiscoCloud only. Other cloud environments
+are set up with the IP in IP mode.
+
+Mesos
+^^^^^
+
+The systemd unit file for Mesos slave is updated with the new ``DOCKER_HOST``
+environment variable:
+
+.. code-block:: shell
+
+   DOCKER_HOST=localhost:2377
+
+This allows Calico to set up networking automatically by routing Docker API
+requests through the `Powerstrip <https://github.com/clusterhq/powerstrip>`_
+proxy that is running on port ``2377`` on each Mesos slave host. It sets
```

Comment:
The Powerstrip plugin was deprecated in Calico starting from [version 0.5.0](https://github.com/Metaswitch/calico-docker/releases/tag/v0.5.0) The version 0.5.0 should be used with Docker version 1.8 which is currently experimental. That is why we use version 0.4.8 which is for the stable Docker version.
I haven't tested the compatibility with version 1.7.1 but will do that shortly.